code/modules/mining/lavaland/seismicarm.dm

```diff
@@ -1,165 +1,170 @@
+	if(target.mobility_flags & MOBILITY_STAND)
+		animate(target, transform = null, time = 0.4 SECONDS, loop = 0)
+
+/datum/action/cooldown/seismic/IsAvailable(feedback = FALSE)
+	. = ..()
+	if(!isliving(owner))
 		return FALSE
-	if(islava(turf_to_check))
+	if(HAS_TRAIT(owner, TRAIT_PACIFISM))
 		return FALSE
-	for(var/obj/object in turf_to_check.contents)
-		if(object.density)
-			return FALSE
-	return TRUE
 
-#define LARIAT_JUMP_DISTANCE 4
-
-/datum/action/cooldown/spell/pointed/seismic/lariat
+/datum/action/cooldown/seismic/lariat
 	name = "Lariat"
 	desc = "Dash forward, catching whatever animal you hit with your arm and knocking them over."
 	button_icon = 'icons/mob/actions/actions_arm.dmi'
 	button_icon_state = "lariat"
-
 	cooldown_time = 7 SECONDS
+	var/jumpdistance = 4
 
-/datum/action/cooldown/spell/pointed/seismic/lariat/InterceptClickOn(mob/living/user, params, atom/target)
-	. = ..()
-	if(!.)
-		return FALSE
-	var/turf/turf_ahead = get_step(get_turf(user), user.dir)
-	var/obj/effect/temp_visual/decoy/fading/threesecond/followup_effect = new(get_turf(user), user)
-	user.visible_message(span_warning("[user] sprints forward with [user.p_their()] arm outstretched!"))
-	playsound(user,'sound/effects/gravhit.ogg', 20, TRUE)
-	for(var/i = 0 to LARIAT_JUMP_DISTANCE)
-		if(!check_turf(turf_ahead))
-			return TRUE //TRUE as in: start cooldown
-		user.forceMove(turf_ahead)
-		walk_towards(followup_effect, user, 0, 1.5)
-		animate(followup_effect, alpha = 0, color = "#00d9ff", time = 0.3 SECONDS)
-		for(var/mob/living/assaulted in turf_ahead.contents)
-			if(assaulted != user)
-				user.forceMove(get_turf(assaulted))
-				to_chat(assaulted, span_userdanger("[user] catches you with [user.p_their()] arm and clotheslines you!"))
-				user.visible_message(span_warning("[user] hits [assaulted] with a lariat!"))
-				assaulted.SpinAnimation(0.5 SECONDS, 1)
-				if(isanimal(assaulted))
-					assaulted.adjustBruteLoss(50)
-				if(iscarbon(assaulted))
-					assaulted.adjustBruteLoss(10)
-				if(issilicon(assaulted))
-					assaulted.adjustBruteLoss(12)
-			playsound(assaulted,'sound/effects/meteorimpact.ogg', 60, 1)
-		turf_ahead = get_step(turf_ahead, user.dir)
-	
+/datum/action/cooldown/seismic/lariat/Activate()
+	var/turf/T = get_step(get_turf(owner), owner.dir)
+	var/turf/Z = get_turf(owner)
+	var/obj/effect/temp_visual/decoy/fading/threesecond/F = new(Z, owner)
+	owner.visible_message(span_warning("[owner] sprints forward with [owner.p_their()] arm outstretched!"))
+	StartCooldown()
+	playsound(owner,'sound/effects/gravhit.ogg', 20, 1)
+	for(var/i = 0 to jumpdistance)
+		if(T.density)
+			return
+		for(var/obj/D in T.contents)
+			if(D.density == TRUE)
+				return
+		for(var/turf/closed/indestructible/I in T.contents)
+			return
+		if(istype(T, /turf/open/lava) || (istype(T, /turf/open/chasm)))
+			return
+		for(var/obj/machinery/door/window/E in Z.contents)
+			if(E.density == TRUE)
+				return 
+		if(T)
+			sleep(0.1 SECONDS)
```

Comment:
why bring back your old code when i refactored everything for you that makes it not require sleep

Reply:
mop wasn't intended to be instant otherwise i wouldn't have it give users the damage reducing buff

Reply:
It could be a timer or a do after